### PR TITLE
Remove safe deployment

### PR DESCRIPTION
**Pull request by Adamantios**
## Proposed changes

Removes the safe deployment abci.

## Fixes

n/a

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [x] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

n/a

### CODE REVIEW

File: packages/valory/agents/oracle/tests/test_oracle.py [hunk 1]
```diff
@@ -120,53 +119,66 @@ class TestABCIPriceEstimationSingleAgent(
...
 @pytest.mark.e2e
 @pytest.mark.parametrize("nb_nodes", (2,))
-class TestABCIPriceEstimationTwoAgents(
-    BaseTestEnd2EndExecution,
-    UseGnosisSafeHardHatNet,
-):
+@pytest.mark.skip(
+    "Enable and set the correct multisig and key pairs "
+    "when registries image gets updated to include a service with 2 agent instances"
+)
```
Comment (Adamantios) [inline packages/valory/agents/oracle/tests/test_oracle.py:159]: This is ready on https://github.com/valory-xyz/autonolas-registries/tree/deploy_service_two_agent_instances. I will create the image, push it, and then re-enable the e2e test on a separate PR.

File: packages/valory/skills/oracle_deployment_abci/rounds.py [hunk 2]
```diff
@@ -67,9 +67,23 @@ def safe_contract_address(self) -> str:
...
+
+    round_id = "ffw"
+    allowed_tx_type = VotingOraclePayload.transaction_type
+    payload_attribute = "vote"
+    synchronized_data_class = SynchronizedData
+    done_event = Event.DONE
+    negative_event = Event.NEGATIVE
+    none_event = Event.NONE
+    no_majority_event = Event.NO_MAJORITY
+    collection_key = "participant_to_ffw"
```
Comment (Adamantios) [inline packages/valory/skills/oracle_deployment_abci/rounds.py:86]: This round is necessary in order to know if the oracle contract address is provided. Now that the safe contract is always provided via the `setup`, the registration startup always ends with a FFW event, so we need to decide whether we need to deploy the oracle contract or not here.

File: packages/valory/skills/oracle_deployment_abci/rounds.py [hunk 3]
```diff
@@ -201,4 +228,4 @@ class OracleDeploymentAbciApp(AbciApp[Event]):
         Event.VALIDATE_TIMEOUT: 30.0,
         Event.DEPLOY_TIMEOUT: 30.0,
     }
-    cross_period_persisted_keys = ["oracle_contract_address"]
+    cross_period_persisted_keys = ["safe_contract_address", "oracle_contract_address"]
```
Comment (0xArdi) [inline packages/valory/skills/oracle_deployment_abci/rounds.py:231]: We could use `get_name` here as well. Also I think we can add the `safe_contract_address` to the cross period keys of registration abci. I don't think we should rely on the users to do this.

Comment (Adamantios) [reply packages/valory/skills/oracle_deployment_abci/rounds.py:231]: Agreed! Though this is not available here yet, it will be done when bumping to open-autonomy `v0.6.0`.

Regarding the `safe_contract_address`, I also think this is a nice idea! One thing to keep in mind is that some users or apps might not need it (such as the APY oracle) and therefore it will be in the cross-period keys without a reason, but I think it is a small trade-off.

**Review by 0xArdi — APPROVED** [review]

File: packages/valory/skills/oracle_deployment_abci/payloads.py [hunk 4]
```diff
@@ -99,13 +99,13 @@ def data(self) -> Dict:
         return dict(keeper=self.keeper)
 
 
-class ValidateOraclePayload(BaseTxPayload):
-    """Represent a transaction payload of type 'validate'."""
+class VotingOraclePayload(BaseTxPayload):
```
Comment (DavidMinarsch) [inline packages/valory/skills/oracle_deployment_abci/payloads.py:102]: why the change?

**Review by DavidMinarsch — COMMENTED** [review]
Suggest we leave open and apply new PRs on top until all of 0.7.0 is available

File: packages/valory/skills/oracle_deployment_abci/payloads.py (hunk 4 above)
Comment (Adamantios) [reply packages/valory/skills/oracle_deployment_abci/payloads.py:102]: I made the payload re-usable on a52dcf75115450df72b04421747205257d236d6c so that it can be re-used by the new round introduced on 40466c228ea308d895a06c20e6ff13d28c3564e0.

A `VotingPayload` should be implemented on the core at some point since it is always the same.

**Comment by Adamantios** [comment]
Third-party hashes cannot remain unchanged because the `0.6.0` version was developed in 2022, however, the copyright script updates the copyright in the third-party packages to 2023, which results in the third-party hashes being updated as well.
